updatehub-sdk/src/listener.rs

```diff
@@ -67,12 +71,29 @@ impl Handler {
     }
 }
 
-impl StateChangeListener {
+impl StateChange {
     #[inline]
     pub fn new() -> Self {
-        StateChangeListener::default()
+        StateChange::default()
     }
 
+    /// Function that register callback(s) for the state(s).
+    /// # Example
+    ///
+    /// ```no_run
+    /// # use updatehub_sdk::{listener, Result};
+    /// # #[async_std::main]
+    /// # async fn main() -> Result<()> {
+    /// # let mut listener = listener::StateChange::default();
+    /// listener.on_state(listener::State::Download, |mut handler| async move {
+    ///     println!("function called when starting the Download state");
+    ///
+    ///     // Cancels the current download.
+    ///     handler.cancel().await
+    /// });
+    /// # listener.listen().await
+    /// # }
+    /// ```
```

Comment:
This block does not produce the expected output, checkout the generated doc page.